### PR TITLE
Enable external DNS in hypershift azure e2e and sets managed-service ARO-HCP

### PR DESCRIPTION
**Pull request by enxebre**

### CODE REVIEW

File: ci-operator/jobs/openshift/linuxptp-daemon/openshift-linuxptp-daemon-release-4.17-presubmits.yaml [hunk 1]
```diff
@@ -5,7 +5,6 @@ presubmits:
     branches:
     - ^release-4\.17$
     - ^release-4\.17-
-    cluster: build03
```
Comment (Patryk-Stefanski) [inline ci-operator/jobs/openshift/linuxptp-daemon/openshift-linuxptp-daemon-release-4.17-presubmits.yaml:8]: We need these

Comment (enxebre) [reply ci-operator/jobs/openshift/linuxptp-daemon/openshift-linuxptp-daemon-release-4.17-presubmits.yaml:8]: not sure why make update got rid of it

**Comment by enxebre** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by enxebre** [comment]
management cluster failed to comeup with some degraded operators. Looking...
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by enxebre** [comment]
external dns was not liking the passed creds

/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by enxebre** [comment]
zone not found. possibly missing perms https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/54024/rehearse-54024-pull-ci-openshift-hypershift-release-4.17-e2e-azure/1809195911469862912/artifacts/e2e-azure/hypershift-azure-run-e2e/artifacts/TestCreateCluster/namespaces/hypershift/core/pods/logs/external-dns-547f6cf747-kg6fr-external-dns.log

/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by enxebre** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by enxebre** [comment]
nodes failed to join. Retesting to debug
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Comment by enxebre** [comment]
needs https://github.com/openshift/hypershift/pull/4336

**Comment by enxebre** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-azure

**Pull request merged**